[Scala 3] Macro leads to class-not-found at runtime when containing a reference to a java enum.

Repro (using [scala-cli](https://scala-cli.virtuslab.org/)) : 

```scala
// using lib org.scala-sbt:test-interface:1.0
// using lib com.eed3si9n.expecty::expecty:0.15.4

import sbt.testing.Status
import com.eed3si9n.expecty.Expecty.assert

@main
def main() = assert(Status.Error == Status.Error)
```

Output : 

```
Compiling project (Scala 3.0.2, JVM)
Compiled project (Scala 3.0.2, JVM)
Exception in thread main: java.lang.Exception: java.lang.NoClassDefFoundError: sbt/testing/Status$
    at method print in modules/runner/src/main/scala-3-stable/scala/cli/runner/Stacktrace.scala:12 inside runner_3.jar 
    at method printException in modules/runner/src/main/scala/scala/cli/runner/StackTracePrinter.scala:91 inside runner_3.jar 
    at method main in modules/runner/src/main/scala/scala/cli/runner/Runner.scala:22 inside runner_3.jar 

Caused by: Exception in thread main: java.lang.ClassNotFoundException: sbt.testing.Status$
    at method loadClass in jdk.internal.loader.BuiltinClassLoader:602 
    at method loadClass in jdk.internal.loader.ClassLoaders$AppClassLoader:178 
    at method loadClass in java.lang.ClassLoader:522 
    at method main in foo.scala:8 
```

Hitting this too:

```scala
//> using scala "3.2.2"
//> using lib "software.amazon.smithy:smithy-model:1.30.0"
//> using lib "com.disneystreaming::weaver-cats:0.8.3"
import software.amazon.smithy.model.shapes.ShapeId
import weaver._

object MyTest extends FunSuite {

  test("demo") {
    val shapes = Set.empty[ShapeId]
    expect(shapes.contains(ShapeId.from("foo#Foo")))
  }

}
```

```
Exception in thread "io-compute-6" java.lang.NoClassDefFoundError: software/amazon/smithy/model/shapes/ShapeId$
        at MyTest$.$init$$$anonfun$1(main.scala:11)
        at scala.util.Try$.apply(Try.scala:210)
        at weaver.Test$.pure(Test.scala:31)
        at weaver.FunSuiteF.test$$anonfun$1(suites.scala:187)
        at weaver.FunSuiteF.pureSpec$$anonfun$1(suites.scala:197)
        at scala.collection.immutable.List.map(List.scala:246)
        at scala.collection.immutable.List.map(List.scala:79)
        at weaver.FunSuiteF.pureSpec(suites.scala:197)
        at weaver.FunSuiteF.spec(suites.scala:201)
        at weaver.EffectSuite.run(suites.scala:52)
        at weaver.EffectSuite.run$(suites.scala:29)
        at weaver.RunnableSuite.run(suites.scala:64)
        at weaver.framework.RunnerCompat$IOTask.$anonfun$9$$anonfun$1$$anonfun$1(RunnerCompat.scala:221)
        at cats.effect.IOFiber.succeeded(IOFiber.scala:1177)
        at cats.effect.IOFiber.runLoop(IOFiber.scala:345)
        at cats.effect.IOFiber.execR(IOFiber.scala:1324)
        at cats.effect.IOFiber.run(IOFiber.scala:118)
        at cats.effect.unsafe.WorkerThread.run(WorkerThread.scala:485)
Caused by: java.lang.ClassNotFoundException: software.amazon.smithy.model.shapes.ShapeId$
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
        ... 18 more
```

--- a/build.sbt
+++ b/build.sbt
@@ -38,7 +38,10 @@ lazy val expecty = (projectMatrix in file("."))
     settings = Seq(
       libraryDependencies += "com.github.sbt" % "junit-interface" % "0.13.3" % Test,
       Test / unmanagedSourceDirectories ++= {
-        Seq((LocalRootProject / baseDirectory).value / "jvm" / "src" / "test" / "scala")
+        Seq(
+          (LocalRootProject / baseDirectory).value / "jvm" / "src" / "test" / "scala",
+          (LocalRootProject / baseDirectory).value / "jvm" / "src" / "test" / "java",
+        )
       },
     )
   )

--- a/jvm/src/test/java/com/example/A.java
+++ b/jvm/src/test/java/com/example/A.java
@@ -1,0 +1,7 @@
+package com.example;
+
+public class A {
+  public static int getNumber() {
+    return 1;
+  }
+}

--- a/jvm/src/test/java/com/example/Level.java
+++ b/jvm/src/test/java/com/example/Level.java
@@ -1,0 +1,7 @@
+package com.example;
+
+public enum Level {
+  LOW,
+  MEDIUM,
+  HIGH
+}

--- a/jvm/src/test/scala/org/expecty/JavaTest.scala
+++ b/jvm/src/test/scala/org/expecty/JavaTest.scala
@@ -1,0 +1,30 @@
+/*
+ * Copyright 2012 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package foo
+
+import com.example.A
+import com.example.Level
+
+object JavaTest extends verify.BasicTestSuite {
+  val assert1 = com.eed3si9n.expecty.Expecty.assert
+
+  test("enum") {
+    assert1(Level.LOW.toString() == "LOW")
+  }
+
+  test("static method") {
+    assert1(A.getNumber() == 1)
+  }
+}

--- a/src/main/scala-3/com/eed3si9n/expecty/RecorderMacro.scala
+++ b/src/main/scala-3/com/eed3si9n/expecty/RecorderMacro.scala
@@ -166,15 +166,19 @@ class RecorderMacro(using qctx0: Quotes) {
         }
       // case TypeApply(x, ys) => recordValue(TypeApply.copy(expr)(recordSubValues(x), ys), expr)
       case TypeApply(x, ys) => TypeApply.copy(expr)(recordSubValues(runtime, x), ys)
-      case Select(x, y) =>
-        if (!x.symbol.flags.is(Flags.Package))
-          Select.copy(expr)(recordAllValues(runtime, x), y)
+      case sel @ Select(x, y) =>
+        if !x.symbol.flags.is(Flags.Package) && !isJavaEnum(x.symbol)
+          && !sel.symbol.flags.is(Flags.JavaStatic)
+        then Select.copy(expr)(recordAllValues(runtime, x), y)
         else expr
       case Typed(x, tpe)   => Typed.copy(expr)(recordSubValues(runtime, x), tpe)
       case Repeated(xs, y) => Repeated.copy(expr)(xs.map(recordAllValues(runtime, _)), y)
       case _               => expr
     }
   }
+
+  private def isJavaEnum(sym: Symbol): Boolean =
+    sym.flags.is(Flags.Enum) && sym.flags.is(Flags.JavaDefined)
 
   private[this] def isImplicitMethod(t: Apply): Boolean = {
     t.symbol.flags.is(Flags.Implicit)

